app/components/avo/views/resource_edit_component.rb

```diff
@@ -34,6 +36,23 @@ def resource_view_path
     helpers.resource_view_path(model: association_resource.model, resource: association_resource)
   end
 
+  def resource_edit_or_new_path
+    modal_id = "new_via_belongs_to"
+
+    if params[:via_resource_id].present?
+      # Back to edit path with param indicating turbo stream should simply close modal
+      related_resource = params[:via_belongs_to_resource_class].constantize.new
+      related_record = related_resource.model_class.find(params[:via_resource_id])
```

Comment:
If we start using `resource.model.to_param` on the `create_path` we need to use [find_record_method](https://docs.avohq.io/2.0/customization.html#custom-find-method-for-show-and-edit-pages) here